### PR TITLE
Setup toolings for API documentation

### LINKED ISSUE #5: Add API documentations
Add documentation to public-facing classes and types. If possible, automate the process of generating documentation.

### PR DESCRIPTION
**Pull request by hawschiat**
## Changes
- Added [@microsoft/api-extractor](https://api-extractor.com/) and [@microsoft/api-documenter](https://www.npmjs.com/package/@microsoft/api-documenter) as dev dependencies to extract and generated API documentation
- Added [eslint-plugin-tsdoc](https://www.npmjs.com/package/eslint-plugin-tsdoc) to catch tsdoc syntax errors
- Added API documentations to `OneSpanSign` class methods and some interfaces (to be finished in separate PRs)
- Removed `OptionalExclude` utility type for more clarity on interfaces

## Next steps
- Set up workflows to automatically generate API reports and documentation (#11 )

### CODE REVIEW

File: etc/onespan-sign-node.api.md [hunk 1]
```diff
@@ -0,0 +1,521 @@
+## API Report File for "onespan-sign-node"
```
Comment (hawschiat) [inline etc/onespan-sign-node.api.md:1]: This API report file is the basis for the markdown files documentation (to be added in separate PRs to keep this PR small)

File: src/types/utils.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
-export type OptionalExclude<T, Exclusion extends keyof T> = Required<Pick<T, Exclusion>> & Partial<Omit<T, Exclusion>>;
```
Comment (hawschiat) [inline src/types/utils.ts:1]: This utility type, although useful, has been removed to make other types more readable and extractable by `api-extractor`

File: README.md [hunk 3]
```diff
@@ -4,6 +4,24 @@
...
+
+Public-facing entities (interfaces, methods, etc.) must be documented properly using [TSDoc](https://tsdoc.org/) syntax. 
+[eslint-plugin-tsdoc](https://www.npmjs.com/package/eslint-plugin-tsdoc) is used to catch improper syntax usage.
+
+API documentation is extracted automatically using [@microsoft/api-extractor](https://api-extractor.com/) and a list of
+supported syntax can be found on their [documentation page](https://api-extractor.com/pages/tsdoc/doc_comment_syntax/).
+
+We use [@microsoft/api-documenter](https://www.npmjs.com/package/@microsoft/api-documenter) to generate documentation.
+To generate documentation locally, you may run the provided script in your terminal:
+
```
Comment (francisco-jara) [inline README.md:17]: might be worth mentioning the prerequisites of installing and building the lib before attempting to generate the docs

**Comment by francisco-jara** [comment]
@hawschiat What's your plan for the generated docs? (e.g. hosting them with Docusaurus?)

**Review by francisco-jara — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (hawschiat) [reply README.md:17]: Do you mean asking the user to install the dependencies first?

Comment (francisco-jara) [reply README.md:17]: and building

**Comment by hawschiat** [comment]
> @hawschiat What's your plan for the generated docs? (e.g. hosting them with Docusaurus?)

@francisco-jara My plan right now is to automatically generate updated documentation when we push changes to `main` (workflow added in #11 ). Users can navigate to the `docs` directory in the repo to view the generated documentation in Markdown files.

**Review by abbieschenk — APPROVED** [review]
